day_10/task_2/pipe_maze.cpp

```diff
@@ -73,31 +74,31 @@ std::pair<MazePoint, MazePoint> AttributedMaze::find_next_neighbors(const MazePo
     const MazePoint current_position(maze_point);
     std::vector<MazePoint> neighbors;
 
-    if(is_connected_to_north(
+    if((current_position.first - 1)< maze_.size() && is_connected_to_north(
```

Comment:
```suggestion
    if(static_cast<std::size_t>(current_position.first - 1) < maze_.size() && is_connected_to_north(
```
1. Add static_cast to make the conversion explicit
2. Add space bar before '<' to make the style consistent.

According to 1. - please watch this: https://www.youtube.com/watch?v=T97QJ0KBaBU to understand the reason
According to 2. - have you applied clang formatter on this PR?

Reply:
done